python/ray/tests/test_scheduling_2.py

```diff
@@ -472,6 +472,29 @@ def get_node_id_task(sleep_s=0):
     assert target_node_id != soft_node_id
 
 
+def test_node_affinity_scheduling_strategy_fail_on_unavailable(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(num_cpus=1)
+    ray.init(address=cluster.address)
+
+    @ray.remote(num_cpus=1)
+    class Actor:
+        def get_node_id(self):
+            return ray.get_runtime_context().get_node_id()
+
+    a1 = Actor.remote()
+    target_node_id = ray.get(a1.get_node_id.remote())
+
+    a2 = Actor.options(
```

Comment:
Is all the combination tested actually? IIUC, the behavior is

```

spill: True fail: True
 -> makes no sense (maybe raise an exception?)

spill:True fail:False

-> spill to other node if other node is available

spill: False fail:True
-> fail if the node is not available

spill:False fail:False
not scheduled until the node is available
```

can you make sure all these scenarios are tested?

Reply:
Currently invalid combinations will check failure since these are private options now and not used by users. Once we make them public, we need to throw proper exceptions. All the valid combinations are tested.